lib/ghomperm.gi

```diff
@@ -1621,15 +1623,47 @@ InstallGlobalFunction( ImageKernelBlocksHomomorphism, function( hom, H,par )
 
             # Make <S> the stabilizer of the block <B>.
             InsertTrivialStabilizer( S.stabilizer, B[ 1 ] );
-            j := 1;
-            while                                j < Length( B )
-                  and Length( S.stabilizer.orbit ) < Length( B )  do
-                j := j + 1;
-                if IsBound( S.translabels[ B[ j ] ] )  then
-                    AddGeneratorsExtendSchreierTree( S.stabilizer,
-                            [ InverseRepresentative( S, B[ j ] ) ] );
-                fi;
-            od;
+
+            if Length(B)>Length(D)^2 then
+              # if there are few, large blocks the search through all block
+              # points is tedious. Rather use an orbit/stabilizer algorithm.
+              orb:=[i];
+              rep:=[One(H)];
+              j:=1;
+              while j<=Length(orb) do
+                for k in S.generators do
+                  img:=D[orb[j]][1]^k;
+                  p:=PositionProperty(D,x->img in x);
+                  if p=fail then Error("block are not well");fi;
+                  if not p in orb then
```

Comment:
So this is yet another ad-hoc orbit algo implementation.

Are the orbits going to be very short? If not, wouldn't it make more sense to use a dictionary here, instead of letting GAP perform a linear search through `orb`.

Reply:
I thought about dictionaries, but it is an orbit on blocks, and I'm always mapping just one point (and then find the block containing the point).
[Just realize that I should be able to remove `PositionProperty` and instead use the stored inverse list -- will do so to make the code better though practical effect will probably be neglegible.]

As for length, the orbit is at most of length |D|, but the condition requires that the permutation degree |B|*|D| is at least |D|^3, that is the orbit is at most third root of permutation degree. 
(Thomas' example that triggered is was degree 150k and action on 9 blocks of size 17k each, this change sped up a calculation including other tasks by a factor of 10)

What might be good is to be more careful still with `AddGenerators', but at this point before a release I don't want to do tricky business whose function is not completely clear.